src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionContainerBase.cs

```diff
@@ -24,11 +25,16 @@ internal abstract class ServiceConnectionContainerBase : IServiceConnectionConta
 
         private static readonly int MaxReconnectBackOffInternalInMilliseconds = 1000;
 
+        private static readonly TimeSpan MessageWriteRetryDelay = TimeSpan.FromMilliseconds(200);
+        private static readonly int MessageWriteMaxRetry = 3;
+
         // Give (interval * 3 + 1) delay when check value expire.
         private static readonly long DefaultServersPingTimeoutTicks = Stopwatch.Frequency * ((long)Constants.Periods.DefaultServersPingInterval.TotalSeconds * 3 + 1);
 
         private static readonly Tuple<string, long> DefaultServersTagContext = new Tuple<string, long>(string.Empty, 0);
 
+        private readonly ConcurrentDictionary<byte, WeakReference<IServiceConnection>> _partitionedCache = new();
```

Comment:
i think we can initialize the dictionary, since we know all of the keys.
and we can use `Dictionary<byte, StrongBox<WeakReference<IServiceConnection>>>`.